PixelHunter1995/Components/SpeechLine.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+
+namespace PixelHunter1995.Components
+{
+    internal class SpeechLine : IUpdateable
+    {
+        private string ActiveSpeech;
+        private TimeSpan SpeechEndTime;
+        public bool Speaking { get; private set; }
+        private bool ShouldStartNextLine;
+        private readonly Queue UpcomingLines;
```

Comment:
Naming gets a bit confusing here as well. `UpcomingLines` are called lines, but when set to being active, they are speeches instead? And the class itself is also a "line". Maybe the `UpcomingLines` should be lines, and `ActiveSpeech` should be `ActiveLine` (or `CurrentLine`, which I like slightly more), and the class `SpeechLine` should be called something else, e.g. `SpeechComponent`?

Reply:
The `SpeechLine` is actually more a manager of lines/speech than an actual line, since the object lives on between lines.

Reply:
(But using `Manager` would feel a bit confusing, since I think `Manager`s should be more on a game level, and not on an individual entity level. That's why I suggested `Component` instead).

Reply:
If you want `SpeechLine` to actually be a line, then you could in the `CharacterComponent` set the initial speech line to a "`NullSpeechLine`" (which might be of its own class, or it might be an instance created with no actual dialog text, and it could be nice to create it through a static function in this class) and when using `Say()` in the `CharacterComponent`, you create a new instance of `SpeechLine` taking the dialog text as a constructor argument.